README.md

```diff
@@ -38,203 +30,404 @@ Add imports at the top of your Noir code, for example:
 use dep::bignum::fields::U256::U256Params;
 use dep::bignum::BigNum;
 ```
+### Addition in U256
 
-## `bignum`
+A simple 1 + 2 = 3 check in 256-bit unsigned integers. Note that for performing multiple arithmetic operations up to degree 2 it is recommended to use `evaluate_quadratic_expression` (see explanation below). 
 
-BigNum members are represented as arrays of 120-bit limbs. The number of 120-bit limbs required to represent a given BigNum object must be defined at compile-time.
+```rust
+use dep::bignum::fields::U256::U256Params;
+use dep::bignum::BigNum;
 
-If your field moduli is _also_ known at compile-time, use the `BigNumTrait` definition in `lib.nr`
+type U256 = BigNum<3, U256Params>;
 
-### BigNum struct
+fn main() {
+    let one: U256 = BigNum::from_array([1, 0, 0]);
+    let two: U256 = BigNum::from_array([2, 0, 0]);
+    let three: U256 = BigNum::from_array([3, 0, 0]);
+    assert((one + two) == three);
+}
+```
 
-Big numbers are instantiated with the BigNum struct:
+## Concepts
+
+### BigNum definition
+
+A BigNum is a number modulo `modulus` and is represented as an array of 120-bit limbs in little endian format. 
 
 ```rust
 struct BigNum<let N: u64, Params> {
     limbs: [Field; N]
 }
 ```
+- `N` is the number of limbs needed to represent the number. Each limb is a `Field`, which contains max 120 bits (a field has 254 bits)
+- `Params` is the parameters associated with the big number. The `modulus` is defined here. 
 
-- `N` is the number of `Field` limbs together holding the value of the big number
-- `Params` is the parameters associated with the big number; refer to sections below for presets and customizations
+This library includes predefined parameter sets. Refer to below for preset info and how to customize the `Params` correctly. 
 
-### Usage
+The actual value of a BigNum can be calculated by multiplying each limb by an increasing power of $2^{120}$. For example `[1,20,300]` represents $1*2^{120*0} + 20*2^{120*1} + 300*2^{120*2}$. We say that the BigNum is represented in radix-$2^{120}$. 
```

Comment:
The `*` does not render, consider `\cdot` for the dot operator.

Reply:
Fixed this with \cdot and an additional space after "radix-"